packages/loopring_v3/circuit/Gadgets/SignatureGadgets.h

```diff
@@ -120,10 +125,15 @@ class CompressPublicKey : public GadgetT {
                               FMT(prefix, ".requireValidPublicKey")),
 
         // Point compression
+
+        // QUESTION(daniel): are we actually just checking publicKeyX > 0 below?
+        // if so, why not use `LtFieldGadget(0, publicKeyX)`?
```

Comment:
These are just field elements so negative values don't really exist. For some reason this is defined here as `(0 - publicKeyX) < publicKeyX`, but `(0 - publicKeyX)` will underflow differently depending on the field. I think basically this checks that `publicKeyX > SNARK_SCALAR_FIELD/2`.